drivers/nutdrv_qx.c

```diff
@@ -3966,12 +3972,12 @@ int	qx_process(item_t *item, const char *command)
 
 	/* Prepare the command to be used */
 	memset(cmd, 0, cmdsz);
-	snprintf(cmd, cmdsz, "%s", command ? command : item->command);
+	snprintf(cmd, cmdsz, "%s%n", command ? command : item->command, &cmd_len);
```

Comment:
Cute trick with `%n`, but why not just check the return value of snprintf (error if negative, number of chars without the ending null-byte otherwise)?

Reply:
Overflow, mainly.

s(n)printf from what I can see returns the theoretical length assuming an infinite buffer, rather than the actual length constrained by the buffer.

Ideally the snprintfs should be memcpys, as the buffer could be either a zero terminated string, or a non zero terminated buffer.

The non-zero terminated buffer part is because of the voltronic's CRC implementation, which is two binary possibly-zero digits before the \r.

This is why we need all the changes to what was until now just a simple string of non-zero characters.